ui/StatusQ/src/StatusQ/Controls/StatusFeeOption.qml

```diff
@@ -0,0 +1,233 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core.Theme 0.1
+
+/*!
+    \qmltype StatusFeeOption
+    \inherits Control
+    \inqmlmodule StatusQ.Controls
+    \since StatusQ.Controls 0.1
+    \brief Displays a clickable option which content is fully customizable
+
+    By design appearance of subText and additionalText as well as their values should be set from outside.
+    If any of those two need to be displayed a component that has an empty value will be rendering a loading state.
+    The control renders based on the selected property.
+    When control is clicked clicked signal will be emitted.
+
+    Example of how to use it:
+
+    \qml
+        StatusFeeOption {
+            subText: "1.65 EUR"
+            showSubText: true
+
+            additionalText: "~40s"
+            showAdditionalText: true
+
+            onSelectedChanged: {
+                // this option is selected/unselected
+            }
+        }
+    \endqml
+
+    For a list of components available see StatusQ.
+ */
+
+Control {
+    id: root
+
+    enum Type {
+        Normal,
+        Fast,
+        Urgent,
+        Custom
+    }
+
+    property int type: StatusFeeOption.Type.Normal
+
+    property bool selected: false
+
+    property string mainText: {
+        switch(type) {
+        case StatusFeeOption.Type.Fast:
+            return qsTr("Fast")
+        case StatusFeeOption.Type.Urgent:
+            return qsTr("Urgent")
+        case StatusFeeOption.Type.Custom:
+            return qsTr("Custom")
+        case StatusFeeOption.Type.Normal:
+        default:
+            return qsTr("Normal")
+        }
+    }
+
+    property string subText
+    property bool showSubText
+
+    property string additionalText
+    property bool showAdditionalText
+
+    property string unselectedText
+
+    property string icon: {
+        switch(type) {
+        case StatusFeeOption.Type.Fast:
+            return Theme.png("wallet/car")
+        case StatusFeeOption.Type.Urgent:
+            return Theme.png("wallet/rocket")
+        case StatusFeeOption.Type.Custom:
+            return Theme.png("wallet/handwrite")
+        case StatusFeeOption.Type.Normal:
+        default:
+            return Theme.png("wallet/clock")
+        }
+    }
+
+    signal clicked()
+
+    font.family: Theme.baseFont.name
+    font.pixelSize: 12
+
+    horizontalPadding: 8
+    verticalPadding: 8
+
+    component AnimatedText: StatusBaseText {
```

Comment:
Careful with these inline components; they tend to crash the `ui-tests` (due to the old Qt version on Ci :(

Reply:
And it does...

```
12:50:17  3/3 Test #3: QmlTests .........................***Exception: SegFault121.64 sec
```

Reply:
Well, it will still crash if you use the inline `component` here :D

Reply:
I had no idea that inline components may cause a crash. :(
I moved it to a stand-alone component, hope the tests will pass now.
@caybro thanks.

Reply:
Yeah they do due to a Qt bug; which is fixed in newer Qt versions

Reply:
But strangely enough, only in tests; the app itself will be fine